packages/mobile/src/harmony-native/components/CompletionCheck/CompletionCheck.tsx

```diff
@@ -0,0 +1,94 @@
+import type { CompletionCheckProps } from '@audius/harmony'
+import styled, { css } from '@emotion/native'
+import type { ViewProps } from 'react-native'
+import Animated, {
+  Easing,
+  interpolate,
+  useAnimatedStyle,
+  useDerivedValue,
+  withTiming
+} from 'react-native-reanimated'
+
+import {
+  Flex,
+  IconValidationCheck,
+  IconValidationX
+} from '@audius/harmony-native'
+
+const CompletionIconBase = styled(Animated.View)({
+  position: 'absolute',
+  justifyContent: 'center',
+  alignItems: 'center',
+  height: 16,
+  width: 16,
+  borderRadius: 9999
+})
+
+const CompletionDefault = styled(CompletionIconBase)(({ theme }) => ({
+  borderWidth: 1,
+  borderStyle: 'solid',
+  borderColor: theme.color.neutral.neutral
+}))
+
+// NOTE: Probably not needed, but leaving code in case we need this later
+// const CompletionEmpty = (props: ViewProps) => <CompletionIconBase {...props} />
+
+const CompletionError = (props: ViewProps) => (
+  <CompletionIconBase {...props}>
+    <IconValidationX />
+  </CompletionIconBase>
+)
+
+const CompletionSuccess = (props: ViewProps) => (
+  <CompletionIconBase {...props}>
+    <IconValidationCheck />
+  </CompletionIconBase>
+)
+
+const useIconValueWithTiming = (
+  value: CompletionCheckProps['value'],
+  iconValue: CompletionCheckProps['value']
+) => {
+  return useDerivedValue(() => {
+    return withTiming(value === iconValue ? 1 : 0, {
+      duration: 180,
+      easing: Easing.inOut(Easing.ease)
+    })
+  })
+}
+
+const useIconAnimation = (value: Animated.SharedValue<0 | 1>) => {
```

Comment:
lets refactor this so the individual completion icons implement all the animation logic, and completion check wrapper just passed down the value and the complete status to each. then i think we can avoid having to make two custom hooks, it can all just live in the icon component